sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,656 @@
+   */
+  kDifPinmuxPadKindMio = 0,
+  /**
+   * Dedicated Input/Output pad. Connected directly to a peripheral input
+   * or output, bypassing the multiplexer matrix.
+   */
+  kDifPinmuxPadKindDio,
+} dif_pinmux_pad_kind_t;
+
+/**
+ * Pin Multiplexer functionality locking.
+ *
+ * Represents a target functionality to be locked for a given
+ * pad/peripheral input.
+ */
+typedef enum dif_pinmux_lock_target {
+  /**
+   * Lock MIO pad input select for a given peripheral input. This means that
+   * the peripheral input connection can no longer be changed.
+   */
+  kDifPinmuxLockTargetInsel = 0,
+  /**
+   * Lock peripheral output select for a given MIO pad output. This means
+   * that the MIO pad output connection can no longer be changed.
+   */
+  kDifPinmuxLockTargetOutsel,
+  /**
+   * Lock the sleep mode configuration for a given MIO pad.
+   */
+  kDifPinmuxLockTargetMioSleep,
+  /**
+   * Lock the sleep mode configuration for a given DIO pad.
+   */
+  kDifPinmuxLockTargetDioSleep,
+  /**
+   * Lock physical characteristics configuration for a given MIO pad.
+   */
+  kDifPinmuxLockTargetMioPadAttr,
+  /**
+   * Lock physical characteristics configuration for a given DIO pad.
+   */
+  kDifPinmuxLockTargetDioPadAttr,
+  /**
+   * Lock wake-up detector configuration for a given detector.
+   */
+  kDifPinmuxLockTargetWakeupDetector,
+
+} dif_pinmux_lock_target_t;
+
+/**
+ * Pin Multiplexer generic index.
+ *
+ * The meaning of this index changes between APIs, and must be documented in
+ * the corresponding function description.
+ */
+typedef uint32_t dif_pinmux_index_t;
+
+/**
+ * Pin multiplexer Padring pad attributes.
+ *
+ * This `enum` is a bitfield, meaning that it is allowed to pass combined value
+ * of multiple individual attributes.
+ *
+ * Some attributes have defined meanings, others are implementation defined. Not
+ * all pads will support all attributes, or attribute combinations. The Pin
+ * Multiplexer DIF does not check for conflicting or unsupported attributes,
+ * however the underlying hardware prevents these from being enabled. If the
+ * combination of attributes is not supported - some/all of these attributes
+ * will still be disabled (relevant bit unset in the attribute field).
+ *
+ * NOTE:
+ * The functions are used to toggle these attributes, must signal an error if at
+ * least one of the attributes in the bitfield could not be toggled.
+ */
+typedef enum dif_pinmux_pad_attr {
+  kDifPinmuxPadAttrNone = 0,
+  kDifPinmuxPadAttrIOInvert = 1 << 0,
+  kDifPinmuxPadAttrVirtOpenDrain = 1 << 1,
+  kDifPinmuxPadAttrPullDown = 1 << 2,
+  kDifPinmuxPadAttrPullUp = 1 << 3,
+  kDifPinmuxPadAttrKeeper = 1 << 4,
+  kDifPinmuxPadAttrWeakestDrive = 1 << 5,
+  kDifPinmuxPadAttrWeakDrive = 1 << 6,
+  kDifPinmuxPadAttrStrongDrive = 1 << 7,
+  kDifPinmuxPadAttrStrongestDrive = 1 << 8,
+  kDifPinmuxPadAttrSchmittTrigger = 1 << 9,
+  kDifPinmuxPadAttrSlewRateSlow = 1 << 10,
+  kDifPinmuxPadAttrSlewRateFast = 1 << 11,
```

Comment:
As with drive strength slew rate must be one or slow or fast. What happens when neither is specified?

Reply:
Same as comment above.